bundle/src/test/java/com/adobe/acs/commons/packaging/impl/AssetPackagerServletImplTest.java

```diff
@@ -0,0 +1,207 @@
+
+    private final String[] assetPrefixResults = {"/content/we-retail/language-masters/en/experience",
+            "/content/dam/we-retail/en/experiences/arctic-surfing-in-lofoten/arctic-surfing-in-lofoten",
+            "/content/dam/we-retail/en/experiences/arctic-surfing-in-lofoten/fjord-waves.jpg",
+            "/content/dam/we-retail/en/experiences/arctic-surfing-in-lofoten/surfer-ready-to-go.jpg",
+            "/content/dam/we-retail/en/experiences/arctic-surfing-in-lofoten/surfer-wave-02.jpg",
+            "/content/dam/we-retail/en/experiences/arctic-surfing-in-lofoten/camp-tent.jpg",
+            "/content/dam/we-retail/en/experiences/arctic-surfing-in-lofoten/camp-fire.jpg",
+            "/content/dam/we-retail/en/experiences/arctic-surfing-in-lofoten/northern-lights.jpg",
+            "/content/dam/we-retail/en/experiences/arctic-surfing-in-lofoten/majestic-rainbow.jpg",
+            "/content/dam/we-retail/en/experiences/arctic-surfing-in-lofoten/surfer-wave-01.jpg",
+            "/content/dam/we-retail/en/experiences/48-hours-of-wilderness/48-hours-of-wilderness-1.jpg"};
+
+    @Before
+    public void setup() {
+        context.load().json(getClass().getResourceAsStream("AssetPackagerServletContent.json"), "/content/we-retail/language-masters/en/experience");
+        context.load().json(getClass().getResourceAsStream("AssetPackagerServletDamContent.json"), "/content/dam/we-retail/en");
+        context.load().json(getClass().getResourceAsStream("AssetPackagerServletPackagesContent.json"), "/etc/packages");
+
+        request = new MockSlingHttpServletRequest(context.resourceResolver(), context.bundleContext());
+        response = context.response();
+
+        doReturn(packageHelper).when(assetPackagerServlet).getPackageHelper();
+        when(packaging.getPackageManager(any(Session.class))).thenReturn(new JcrPackageManagerImpl(context.resourceResolver().adaptTo(Session.class)));
+    }
+
+    @Test
+    public void testPreview() throws Exception {
+        context.load().json(getClass().getResourceAsStream("AssetPackagerServletConfiguration.json"), PACKAGER_CONTENT_PATH);
+        request.setResource(context.resourceResolver().getResource(PACKAGER_JCR_CONTENT_PATH));
+        Map<String, Object> params = new HashMap<>();
+        params.put(PREVIEW, true);
+        request.setParameterMap(params);
+        assetPackagerServlet.doPost(request, response);
+        JSONObject responseJSON = new JSONObject(response.getOutputAsString());
+        assertEquals(responseJSON.getString(STATUS), PREVIEW);
+        JSONArray filterSets = responseJSON.getJSONArray(FILTER_OBJECT);
+        assertEquals(filterSets.length(), previewResults.length);
+        for (int i = 0; i < previewResults.length; i++) {
+            assertEquals(filterSets.getJSONObject(i).getString(ROOT_PATH), previewResults[i]);
+        }
+    }
+
+    @Test
+    public void testAssetExclusion() throws Exception {
```

Comment:
Would it make sense for these tests to validate that multiple exclusion paths are honored?

Reply:
I've updated the tests to include 2 tests for multiple exclusions. Something I would like to test but can't figure out a way to do is to test if the filter in the package _actually has_ the exclusion. I can assume that it does because the assets on that page are not present in the JSON response, but I can't verify the actual filters in it. See screens:
JSON response from servlet:
<img width="709" alt="Screen Shot 2019-05-13 at 4 57 21 PM" src="https://user-images.githubusercontent.com/20479202/57659653-4b4c0080-75a0-11e9-82d5-53e188c8a270.png">
Actual filters of the package:
<img width="830" alt="Screen Shot 2019-05-13 at 4 57 37 PM" src="https://user-images.githubusercontent.com/20479202/57659669-569f2c00-75a0-11e9-8c74-e90d5fc67c6e.png">

I don't feel _that_ bad about testing this though because technically it should be part of the PackageHelperImplTest class :)